packages/teams-js/src/public/search.ts

```diff
@@ -0,0 +1,131 @@
+   * will be whatever the last query was before ending search.
+   * @param onExecuteHandler - This optional handler will be called whenever the user 'executes' the
+   * search (by pressing enter for example). The value of {@link SearchQuery.searchTerm} is the current 
+   * term the user is searching for. Should be used if your app wants to treat executing searches differently than responding
+   * to changes to the search query.
+   *
+   * @example
+   * ``` ts
+   * search.registerHandlers(
+      query => {
+        console.log(`Update your application with the changed search query: ${query.searchTerm}`);
+      },
+      () => {
+        console.log('Update your application to handle the search experience being closed');
+      },
+      query => {
+        console.log(`Update your application to handle an executed search result: ${query.searchTerm}`);
+      },
+     );
+   * ```
+   */
+  export function registerHandlers(
+    onChangeHandler: SearchQueryHandler,
+    onClosedHandler: SearchQueryHandler,
+    onExecuteHandler?: SearchQueryHandler,
+  ): void {
+    // TODO: figure out what frame contexts you want to support this in
+    // This is just a guess that I made and should be something you make
+    // an explicit decision about.
+    ensureInitialized(
+      FrameContexts.content,
+      FrameContexts.task,
+      FrameContexts.sidePanel,
+      FrameContexts.stage,
+      FrameContexts.meetingStage,
+    );
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    registerHandler(onChangeHandlerName, onChangeHandler);
+    registerHandler(onClosedHandlerName, onClosedHandler);
+    if (onExecuteHandler) {
+      registerHandler(onExecutedHandlerName, onExecuteHandler);
+    }
+  }
+
+  /**
+   * Allows the caller to unregister for all events fired by the host search experience. Calling
+   * this function will cause your app to stop appearing in the set of search scopes in the hosts
+   */
+  export function unregisterHandlers(): void {
+    // TODO: figure out what frame contexts you want to support this in
+    // This is just a guess that I made and should be something you make
+    // an explicit decision about.
+    ensureInitialized(
+      FrameContexts.content,
+      FrameContexts.task,
+      FrameContexts.sidePanel,
+      FrameContexts.stage,
+      FrameContexts.meetingStage,
+    );
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    // This should let the host know to stop making the app scope show up in the search experience
+    // Can also be used to clean up handlers on the host if desired
+    sendMessageToParent('search.unregister');
+    removeHandler(onChangeHandlerName);
+    removeHandler(onClosedHandlerName);
+    removeHandler(onExecutedHandlerName);
+  }
+
+  /**
+   * Checks if search capability is supported by the host
+   * @returns true if the search capability is enabled in runtime.supports.search and
```

Comment:
Thinking about this comment more (I assume I wrote it), we should probably change it to be:

`@returns true if the search capability is supported by the host and false otherwise`

The information about runtime.supports is useful for you know when implementing but not something other developers will need to know